### PR TITLE
Aggressive refactoring and support Python 3

### PR DESCRIPTION
**Pull request by lambdalisue**
Thank you for this great plugin. It is simple and clear, I wonder why I didn't create this kind of plugin before.

Anyway, I would like to use this plugin in python3 so I did aggressive refactoring and support python3.
I sent PR but I change a lot of things which don't need to be changed as well, so this PR is like a mention just in case if you like that changes.

Best.

### CODE REVIEW

**Comment by cxhernandez** [comment]
Thanks for your interest in the code and this PR, @lambdalisue!

Are you sure these changes are py3 compatible though? To my knowledge, the `xmlrpclib` package has been replaced by `xmlrpc`. Having to learn this new API is what prevented me from including py3 compatibility in the first place :confused:...

File: src/ipymol/core.py [hunk 1]
```diff
@@ -2,44 +2,46 @@
...
+        self.port = int(port)
+        self._thread = threading.Thread(
+            target=os.system,
+            args=(('pymol -cKRQ',)),
+        )
+        self._thread.daemon = True
+
+    def start(self):
+        if self._thread.is_alive():
+            print "A PyMOL RPC server is already running."
```
Comment (cxhernandez) [inline src/ipymol/core.py:26]: The should be a print function, not a statement.

**Comment by lambdalisue** [comment]
That kind of things will automatically be translated via setup.py

File: src/ipymol/core.py (hunk 1 above)
Comment (lambdalisue) [inline src/ipymol/core.py:26]: automatically be translated

**Comment by lambdalisue** [comment]
use2to3 in setup.py does the trick

**Comment by lambdalisue** [comment]
Read this. https://pythonhosted.org/setuptools/python3.html

**Comment by lambdalisue** [comment]
And note that your version said MIT
or Apache license and I simply set everything MIT. So please collect if its wrong.

**Comment by lambdalisue** [comment]
Ok I'm back. Sorry for the short messages, I was in iPhone.

First of all, `use_2to3` in `setup.py` will execute `2to3` command which was introduced when Python 3 was introduced. The command will automatically translate most of incompatible features in Python 2 (such as `print 'xxx'` or renamed modules) into Python 3 style. Thus you don't have to make most of code compatible to Python 3 (I said 'most' because several features are incompatible even with `2to3`, but in this case, everything seems compatible with `2to3`).

So what you need is install IPyMOL via `setup.py` or `pip` or `easy_install` like

```
$ python3 setup.py install
$ python3 setup.py develop
$ pip3 install .
```

Then all actual files installed in python package will automatically be translated to Python 3 style.

See the top right said 'Python 3'.

**Comment by cxhernandez** [comment]
I see! Sorry– when I first installed the code I got an error that `xmlrpclib` couldn't be imported, so I thought this was something `use2to3` couldn't handle. Thanks for the clarification!

The MIT license is fine!

**Comment by lambdalisue** [comment]
I'm glad to make this compatible. Thanks for this great plugin :smile: 

One more thing. It is better if I can install this via `pip install ipymol` thus could you register the package to PyPI?

```
$ python setup.py register
$ python setup.py sdist upload
```

**Comment by cxhernandez** [comment]
Done! Thanks again!

**Comment by lambdalisue** [comment]
Thanks!

**Comment by lambdalisue** [comment]
Oops sorry, I'll send PR.

**Pull request merged**